### PR TITLE
[#116] Supports GHC 9.6.1

### LINKED ISSUE #116: Support GHC 9.6
The GHC 9.6.1 was released recently and it makes sense to support compilation of `iris` with the new version of the Haskell compiler.

Steps needed to be done:

- [ ] Increase `base` upper bound in `iris.cabal`
    https://github.com/chshersh/iris/blob/e9b6c25df2bf179dc8d9f32f05c55aa8f7febcce/iris.cabal#L29-L30

- [ ] Update the `tested-with` field
    https://github.com/chshersh/iris/blob/e9b6c25df2bf179dc8d9f32f05c55aa8f7febcce/iris.cabal#L19-L23

- [ ] Add new warnings
    https://github.com/chshersh/iris/blob/e9b6c25df2bf179dc8d9f32f05c55aa8f7febcce/iris.cabal#L55-L56

- [ ] Fix all compilation errors and warnings if any

- [ ] Update CI pipelines
    > Refer to the following blog post for guidance around CI:
    > 
    > * [Dead simple cross-platform GitHub Actions for Haskell](https://kodimensional.dev/github-actions)

    https://github.com/chshersh/iris/blob/e9b6c25df2bf179dc8d9f32f05c55aa8f7febcce/.github/workflows/ci.yml#L20-L24

### PR DESCRIPTION
**Pull request by celsobonutti**
Resolves #116

Supports GHC 9.6.1 and updates tested GHC 9.2.5 to 9.2.7.

### Additional tasks

- [ ] Documentation for changes provided/changed (N/A)
- [ ] Tests added (N/A)
- [X] Updated CHANGELOG.md

### CODE REVIEW

File: iris.cabal [hunk 1]
```diff
@@ -27,7 +28,7 @@ source-repository head
   location:            https://github.com/chshersh/iris.git
 
 common common-options
-  build-depends:       base >= 4.14 && < 4.18
+  build-depends:       base >= 4.14 && < 4.19
```
Comment (chshersh) [inline iris.cabal:31]: Let's actually bump the upper bound to `< 5`.
```suggestion
  build-depends:       base >= 4.14 && < 5
```

I actually wrote a blog post about this recently 😅 

* https://dev.to/chshersh/new-oss-rule-do-as-little-work-as-possible-4j2k

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,10 @@ Implement Yes/No reading functions:
   * Add `YesNo` type (`Yes` | `No`)
 
   (by [@martinhelmer])
+  
+[#116](https://github.com/chshersh/iris/issues/116):
+Supports GHC 9.6.1
+  (by [@celsobonutti])
```
Comment (chshersh) [inline CHANGELOG.md:24]: Minor formatting changes.

Also, could you please your GitHub username to the `<!-- Contributors -->` section further below in this file?

```suggestion
* [#116](https://github.com/chshersh/iris/issues/116):
  Supports GHC 9.6.1
  (by [@celsobonutti])
```

**Review by chshersh — CHANGES_REQUESTED** [review]
Everything looks great!

Also nice to see that CI is successfully passing, and no code changes are required 🚀 

I have a few minor suggestions but overall it looks nice 👌🏻

File: examples/simple-grep/simple-grep.cabal [hunk 3]
```diff
@@ -4,7 +4,7 @@ version:             0.0.0.0
 build-type:          Simple
 
 common common-options
-  build-depends:       base >= 4.14 && < 4.18
+  build-depends:       base >= 4.14 && < 4.19
```
Comment (chshersh) [inline examples/simple-grep/simple-grep.cabal:7]: And here too
```suggestion
  build-depends:       base >= 4.14 && < 5
```

**Comment by celsobonutti** [comment]
Feedback applied! I'll take a look at the blog post 😊

**Review by chshersh — APPROVED** [review]
Thanks a lot! Looks perfect 🤌

**Pull request merged**